### PR TITLE
fix: render navbar megamenu in a portal

### PR DESCRIPTION
**Pull request by karrui**
### TL;DR
Also closes ISOM-1324

Replaced FocusScope with Dialog component for improved accessibility in the NavItem megamenu.

Also fixes stacking issues with other components.

### What changed?

- Replaced `FocusScope` from react-aria with `Dialog` from @headlessui/react in the NavItem component.
- Removed the overlay div from the Navbar component.
- Added a background overlay to the Dialog component.
- Updated the NavbarItemProps interface to include new properties for the Dialog component.

### How to test?

1. Navigate to the Collection story in Storybook.
2. Interact with the navbar items to open and close the megamenu.
3. Verify that the megamenu opens and closes correctly.
4. Check that focus management and keyboard navigation work as expected within the megamenu.
5. Ensure the background overlay appears when the megamenu is open.

### Why make this change?

This change improves accessibility and user experience by:

1. Using a more robust Dialog component for managing focus and keyboard interactions.
2. Providing a consistent background overlay when the megamenu is open.
3. Simplifying the code structure by removing redundant elements and consolidating overlay functionality.

These improvements align with best practices for creating accessible and user-friendly navigation components.

---

### CODE REVIEW

**Comment by karrui** [comment]
* **#482** <a href="https://app.graphite.dev/github/pr/opengovsg/isomer/482?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @karrui and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by karrui** [comment]
wait ah scrolling is broken, it doesn't bring back my scroll lol

**Comment by karrui** [comment]
After this is merged, may be able to remove the style listener on studio, since `usePreventScroll` is removed too.

File: packages/components/src/templates/next/components/internal/Navbar/NavItem.tsx [hunk 1]
```diff
@@ -70,61 +70,61 @@ export const NavItem = forwardRef<HTMLButtonElement, NavbarItemProps>(
...
-                      onClick={onCloseMegamenu}
-                      aria-label="Close navigation item"
-                      className="flex h-[2.125rem] w-[2.125rem] items-center justify-center text-[1.5rem]"
-                    >
-                      <BiX />
-                    </button>
+          <Dialog
+            open={isOpen}
+            onClose={onCloseMegamenu}
+            className="fixed inset-0 bg-canvas-overlay/40"
```
Comment (dcshzj) [inline packages/components/src/templates/next/components/internal/Navbar/NavItem.tsx:76]: issue(non-blocking): Clicking on the nav item to close it doesn't actually close it.

https://github.com/user-attachments/assets/9b876119-3e0e-492a-90f3-cbc0c0533ba9

But okay we can settle in a separate PR.

**Review by dcshzj — APPROVED** [review]

**Pull request merged**